modules/core/src/commonMain/kotlin/fr/acinq/lightning/channel/Helpers.kt

```diff
@@ -304,88 +297,127 @@ object Helpers {
                     Script.isPay2wsh(script) -> true
                     // option_shutdown_anysegwit doesn't cover segwit v0
                     Script.isNativeWitnessScript(script) && script[0] != OP_0 -> allowAnySegwit
+                    script[0] == OP_RETURN -> allowOpReturn
                     else -> false
                 }
             }.getOrElse { false }
         }
 
-        fun isValidFinalScriptPubkey(scriptPubKey: ByteVector, allowAnySegwit: Boolean): Boolean = isValidFinalScriptPubkey(scriptPubKey.toByteArray(), allowAnySegwit)
+        fun isValidFinalScriptPubkey(scriptPubKey: ByteVector, allowAnySegwit: Boolean, allowOpReturn: Boolean): Boolean = isValidFinalScriptPubkey(scriptPubKey.toByteArray(), allowAnySegwit, allowOpReturn)
 
-        private fun firstClosingFee(commitment: FullCommitment, localScriptPubkey: ByteArray, remoteScriptPubkey: ByteArray, requestedFeerate: ClosingFeerates): ClosingFees {
-            // this is just to estimate the weight which depends on the size of the pubkey scripts
-            val dummyClosingTx = Transactions.makeClosingTx(commitment.commitInput, localScriptPubkey, remoteScriptPubkey, commitment.params.localParams.paysClosingFees, 0.sat, 0.sat, commitment.localCommit.spec)
-            val closingWeight = Transaction.weight(Transactions.addSigs(dummyClosingTx, dummyPublicKey, commitment.remoteFundingPubkey, Transactions.PlaceHolderSig, Transactions.PlaceHolderSig).tx)
-            return requestedFeerate.computeFees(closingWeight)
-        }
-
-        fun firstClosingFee(commitment: FullCommitment, localScriptPubkey: ByteVector, remoteScriptPubkey: ByteVector, requestedFeerate: ClosingFeerates): ClosingFees =
-            firstClosingFee(commitment, localScriptPubkey.toByteArray(), remoteScriptPubkey.toByteArray(), requestedFeerate)
-
-        fun nextClosingFee(localClosingFee: Satoshi, remoteClosingFee: Satoshi): Satoshi = ((localClosingFee + remoteClosingFee) / 4) * 2
-
-        fun makeFirstClosingTx(
+        /** We are the closer: we sign closing transactions for which we pay the fees. */
+        fun makeClosingTxs(
             channelKeys: KeyManager.ChannelKeys,
             commitment: FullCommitment,
-            localScriptPubkey: ByteArray,
-            remoteScriptPubkey: ByteArray,
-            requestedFeerate: ClosingFeerates
-        ): Pair<ClosingTx, ClosingSigned> {
-            val closingFees = firstClosingFee(commitment, localScriptPubkey, remoteScriptPubkey, requestedFeerate)
-            return makeClosingTx(channelKeys, commitment, localScriptPubkey, remoteScriptPubkey, closingFees)
+            localScriptPubkey: ByteVector,
+            remoteScriptPubkey: ByteVector,
+            feerate: FeeratePerKw,
+            lockTime: Long,
+        ): Either<ChannelException, Pair<Transactions.ClosingTxs, ClosingComplete>> {
```

Comment:
Having a `logger` here to log our feerate would have been extra nice for testing, but prob not worth the hassle.